shell/platform/darwin/ios/ios_external_texture_gl.mm

```diff
@@ -58,12 +58,13 @@
                                  bool freeze,
                                  GrContext* context) {
   EnsureTextureCacheExists();
-  if (!freeze) {
+  if ( (!freeze && new_frame_ready_ ) || !texture_ref_) {
```

Comment:
Why do we need the `|| !texture_ref_` part? I'm trying to understand the following branches:

1. We are frozen (`freeze = true`) and `texture_ref_` is nil.
2. No new frame is ready and `texture_ref_` is nil.

In both these cases wouldn't we always have an older texture generated from previous frame's pixel buffer?

Reply:
When app resigns and enters background, `texture_ref_` resets to nil in `OnGrContextDestroyed`. Then, app becomes active, `freeze` is `false`, `new_frame_ready_` is `false`, `texture_ref` is nil. 
If `texture_ref_` is nil, the texture rendering output is blank.
In this case, it is needed to regenerate `texture_ref_`.

Reply:
This is useful. Could you factor this conditional into a method and document this behavior? We would ideally have a test for this, but since that is currently hard, lets document this behavior.